File: gerador.java
import java.util.*;

public class gerador {
    public static void main(String[] args) {

        Random rand = new Random();
        List<Integer> caracteresEvitados = Arrays.asList(34, 39, 40, 44, 46, 41, 47, 58, 91, 92, 93, 94, 96);
        Scanner sc = new Scanner(System.in);
        int quant, numeroAleatorio;

        System.out.println("Quantos caracteres sua senha deve conter? ");
        quant = sc.nextInt();

        char[] vet = new char[quant];

        for (int i = 0; i < quant; i++) {
            do {
                // Gerar um número aleatório entre 33 e 122
                numeroAleatorio = rand.nextInt(90) + 33;
            } while (caracteresEvitados.contains(numeroAleatorio)); // Verifica se o número gerado está na lista de caracteres evitados

            char caractere = (char) numeroAleatorio;
            
            vet[i] = caractere;
        }

        System.out.print("Senha gerada: ");

        for (int i = 0; i < quant; i++) {
            System.out.print(vet[i]);
        }
        System.out.println();
    }
}

File: senha-python/gerador.py
import random


quant = int(input("Quantos digitos deve possuir sua senha ? "))

vet = [0 for x in range(quant)]

caracteres_evevitados = [34, 39, 40, 44, 46, 41, 47, 58, 91, 92, 93, 94, 96]

for i in range(0, quant):
    vet[i] = random.randint(33, 122)
    while vet[i] == caracteres_evevitados:
        vet[i] = random.randint(33, 122)
    

print("Senha gerada: ", end="")

for i in range(0, quant):
    print(chr(vet[i]), end="")

